AnaTutorial/src/AnaTutorial.cc

```diff
@@ -0,0 +1,849 @@
+#include "AnaTutorial.h"
+
+/// Cluster/Calorimeter includes
+#include <calobase/RawCluster.h>
+#include <calobase/RawClusterContainer.h>
+#include <calobase/RawClusterUtility.h>
+#include <calobase/RawTower.h>
+#include <calobase/RawTowerContainer.h>
+#include <calobase/RawTowerGeom.h>
+#include <calobase/RawTowerGeomContainer.h>
+#include <calotrigger/CaloTriggerInfo.h>
+
+/// Jet includes
+#include <g4jets/Jet.h>
+#include <g4jets/JetMap.h>
+
+/// Tracking includes
+#include <g4vertex/GlobalVertex.h>
+#include <g4vertex/GlobalVertexMap.h>
+#include <trackbase_historic/SvtxTrack.h>
+#include <trackbase_historic/SvtxTrackMap.h>
+#include <trackbase_historic/SvtxVertex.h>
+#include <trackbase_historic/SvtxVertexMap.h>
+
+/// Truth evaluation includes
+#include <g4eval/JetEvalStack.h>
+#include <g4eval/SvtxEvalStack.h>
+
+/// HEPMC truth includes
+#include <HepMC/GenEvent.h>
+#include <HepMC/GenVertex.h>
+#include <phhepmc/PHHepMCGenEvent.h>
+#include <phhepmc/PHHepMCGenEventMap.h>
+
+/// Fun4All includes
+#include <fun4all/Fun4AllHistoManager.h>
+#include <fun4all/Fun4AllReturnCodes.h>
+#include <g4main/PHG4Hit.h>
+#include <g4main/PHG4Particle.h>
+#include <g4main/PHG4TruthInfoContainer.h>
+#include <phool/PHCompositeNode.h>
+#include <phool/getClass.h>
+
+/// ROOT includes
+#include <TFile.h>
+#include <TH1.h>
+#include <TH2.h>
+#include <TMath.h>
+#include <TNtuple.h>
+#include <TTree.h>
+
+/// C++ includes
+#include <cassert>
+#include <sstream>
+#include <string>
+
+using namespace std;
+
+/**
+ * This class demonstrates the construction and use of an analysis module 
+ * within the sPHENIX Fun4All framework. It is intended to show how to 
+ * obtain physics objects from the analysis tree, and then write them out
+ * to a ROOT tree and file for further offline analysis.  
+ */
+
+/**
+ * Constructor of module
+ */
+AnaTutorial::AnaTutorial(const std::string &name, const std::string &filename)
```

Comment:
Few other cppcheck warnings:
```
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_outfile' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_clustertree' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_tracktree' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_hepmctree' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_truthtree' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_recojettree' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_truthjettree' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_phi_h' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_eta_phi_h' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_clusenergy' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_cluseta' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_clustheta' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_cluspt' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_clusphi' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_cluspx' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_cluspy' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_cluspz' is not initialized in the constructor.
[AnaTutorial.cc:69]: (warning) Member variable 'AnaTutorial::m_E_4x4' is not initialized in the constructor.
```

Reply:
I will fix this, but my cppcheck doesn't give all of these warnings? I don't know why...
```
Singularity rhic_sl7_ext.simg:~/ORNL/sPHENIX/git/tutorials/AnaTutorial/src> cppcheck --enable=all AnaTutorial.cc
Checking AnaTutorial.cc ...
[AnaTutorial.cc:164]: (style) The function 'End' is never used.
[AnaTutorial.cc:101]: (style) The function 'Init' is never used.
[AnaTutorial.cc:126]: (style) The function 'process_event' is never used.
(information) Cppcheck cannot find all the include files (use --check-config for details)
```

Reply:
Regardless, these should all be fixed by moving the `initialize` functions into the constructor as you previously suggested

Reply:
I was running 
```
cppcheck --enable=all -j6 --std=c++11 --language=c++ ./
```